vulcanai/models/basenetwork.py

```diff
@@ -665,14 +665,16 @@ def fit(self, train_loader, val_loader, epochs,
                     self.lr_scheduler.step(epoch=epoch)
 
                 if self.early_stopping:
-                    if self.early_stopping_metric == "loss":
-                        if valid_loss:
+
+                    # only call if you've actually measured these.
+                    if valid_loss:
+                        if self.early_stopping_metric == "loss":
                             valid_loss_neg = -1 * valid_loss
-                        early_stopping(valid_loss_neg, self)
-                    elif self.early_stopping_metric == "accuracy":
-                        early_stopping(valid_acc, self)
-                    else:
-                        raise ValueError("Not a valid stopping metric")
+                            early_stopping(valid_loss_neg, self)
+                        elif self.early_stopping_metric == "accuracy":
+                            early_stopping(valid_acc, self)
+                        else:
+                            raise ValueError("Not a valid stopping metric")
```

Comment:
I would briefly also return in the error the two types of valid metrics to use for this to make the error more actionable

Reply:
done